system/include/emscripten/threading.h

```diff
@@ -255,43 +255,49 @@ EMSCRIPTEN_RESULT emscripten_wait_for_call_i(em_queued_call *call, double timeou
 
 void emscripten_async_waitable_close(em_queued_call *call);
 
-// Runs the given function on the specified thread. If we are currently on
-// that target thread then we just execute the call synchronously; otherwise it
-// is queued on that thread to execute asynchronously.
-// Returns 1 if it executed the code (i.e., it was on the target thread), and 0
-// otherwise.
+// Runs the given function on the specified thread. If we are currently on that
+// target thread then we just execute the call synchronously; otherwise it is
+// queued on that thread to execute asynchronously. Returns 1 if the function
+// was successfully run or dispatched and 0 otherwise.
+int emscripten_dispatch_to_thread_ptr(pthread_t target_thread,
+                                      void (*func)(void*),
+                                      void* arg);
```

Comment:
Sorry, I am probably missing something.

If I follow properly, this new `_ptr` variant would be used in the form
```c
void myProxiedFunction(void *arg) {
  // do something
}

...

void *someOpaqueArg = ...;
emscripten_dispatch_to_thread_ptr(target_thread, myProxiedFunction, someOpaqueArg);
```

What prevented one from using the existing `emscripten_dispatch_to_thread()` method instead? That would look

```c
emscripten_dispatch_to_thread(target_thread, EM_FUNC_SIG_VI, (void*)myProxiedFunction, someOpaqueArg);
```

I am not sure I see yet why this new `x_ptr()` function family would be needed?

Reply:
Yes, those two calls would be equivalent. Even without adding any new API, I would want to refactor the implementation to have something like `emscripten_dispatch_to_thread_ptr` internally and at that point it seems reasonable to expose it as a public API so that users don't need to deal with the `sig` or `satellite` arguments.

But latest plan described [here](https://github.com/emscripten-core/emscripten/pull/15705#issuecomment-985884887) doesn't require us to change this API at all, so I'll close this PR.